examples/eSealing/src/lib.rs

```diff
@@ -0,0 +1,337 @@
+            timestamp,
+            witness,
+        });
+    }
+}
+
+/// Tagged events to be serialized for the event log.
+#[derive(Debug, Serial, SchemaType)]
+enum Event {
+    Registration(RegistrationEvent),
+}
+
+/// The RegistrationEvent is logged when a new file hash is registered.
+#[derive(Debug, Serialize, SchemaType)]
+pub struct RegistrationEvent {
+    /// Hash of the file to be registered by the witness (sender_account).
+    file_hash: FileHash,
+    /// Witness (sender_account) that registered the above file hash.
+    witness:   AccountAddress,
+    /// Timestamp when this file hash was registered in the smart contract.
+    timestamp: Timestamp,
+}
+
+/// Init function that creates this eSealing smart contract.
+#[init(contract = "eSealing", event = "Event")]
+fn contract_init<S: HasStateApi>(
+    _ctx: &impl HasInitContext,
+    state_builder: &mut StateBuilder<S>,
+) -> InitResult<State<S>> {
+    Ok(State::new(state_builder))
+}
+
+/// Register a new file.
+///
+/// It rejects if:
+/// - It fails to parse the parameter.
+/// - If file hash has already been registered.
+/// - If a smart contract tries to register the file hash.
+#[receive(
+    contract = "eSealing",
+    name = "registerFile",
+    parameter = "FileHash",
+    error = "ContractError",
+    mutable,
+    enable_logger
+)]
+fn register_file<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    host: &mut impl HasHost<State<S>, StateApiType = S>,
+    logger: &mut impl HasLogger,
+) -> Result<(), ContractError> {
+    // Ensure that only accounts can register a file.
+    let sender_account = match ctx.sender() {
+        Address::Contract(_) => bail!(ContractError::OnlyAccount),
+        Address::Account(account_address) => account_address,
+    };
+
+    let file_hash: FileHash = ctx.parameter_cursor().get()?;
+
+    let (state, _) = host.state_and_builder();
+
+    // Ensure that the file hash hasn't been registered so far.
+    ensure!(!state.file_exists(&file_hash), ContractError::AlreadyRegistered);
+
+    let slot_time = ctx.metadata().slot_time();
+
+    // Register the file hash.
+    state.add_file(file_hash, slot_time, sender_account);
+
+    // Log the event.
+    logger.log(&Event::Registration(RegistrationEvent {
+        file_hash,
+        witness: sender_account,
+        timestamp: slot_time,
+    }))?;
+
+    Result::Ok(())
+}
+
+/// Get the timestamp and witness of a registered file hash.
+/// If the file hash has not been registered, this query returns
+/// (Timestamp::from_timestamp_millis(0), AccountAddress([0u8; 32])).
+///
+/// It rejects if:
+/// - It fails to parse the parameter.
+#[receive(
+    contract = "eSealing",
+    name = "getFile",
+    parameter = "FileHash",
+    error = "ContractError",
+    return_value = "(Timestamp, AccountAddress)"
+)]
+fn get_file<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    host: &impl HasHost<State<S>, StateApiType = S>,
+) -> ReceiveResult<(Timestamp, AccountAddress)> {
```

Comment:
I recommend changing this to `Option<(Timestamp, AccountAddress)>` instead.

Reply:
I changed it to `Option<FileState>`